external-resolver/src/test/scala/org/genivi/sota/resolver/test/random/Query.scala

```diff
@@ -103,14 +133,30 @@ object Query extends
       s    <- StateT.stateTMonadState[Gen, RawStore].get
       vehs <- Store.numberOfVehicles
       pkgs <- Store.numberOfPackages
+      cmps <- Store.numberOfComponents
+      flts <- Store.numberOfFilters
+      vcomp <- Store.numberOfVehiclesWithSomeComponent
+      vpaks <- Store.numberOfVehiclesWithSomePackage
+      pfilt <- Store.numberOfPackagesWithSomeFilter
       qry  <- lift(Gen.frequency(
 
         (10, Gen.const(ListVehicles)),
+        (10, Gen.const(ListComponents)),
+        ( 5, Gen.const(ListFilters)),
+
+        (if (vehs > 0) 10 else 0, Gen.oneOf(
+          Store.pickVehicle.runA(s).map(ListPackagesOnVehicle(_)),
+          Store.pickVehicle.runA(s).map(ListComponentsFor(_))
+        )),
+
+        (if (vcomp > 0) 10 else 0,
+          Store.pickVehicleWithComponent.runA(s) map { case (veh, cmp) => ListVehiclesFor(cmp) }),
 
-        (if (vehs > 0) 10 else 0,
-          Store.pickVehicle.runA(s).map(ListPackagesOnVehicle(_))),
+        (if (pfilt > 0) 10 else 0,
+          Store.pickPackageWithFilter.runA(s) map { case (pkg, flt) => ListPackagesFor(flt) }),
 
-        (5,  Gen.const(ListFilters)),
+        (if (pkgs > 0) 10 else 0,
+          Store.pickPackage.runA(s).map(ListFiltersFor(_))),
```

Comment:
I suppose this list will be empty very often. Does it make sense to use pickPackageWithFilter here as well?